static/src/javascripts/projects/common/utils/detect.js

```diff
@@ -71,6 +67,72 @@ define([
         ],
         detect;
 
+    var breakpointNames = breakpoints.map(getBreakpointName);
+
+    var currentBreakpoint;
+    var currentTweakpoint;
+
+    init(window);
+
+    function init(win) {
+        if ('matchMedia' in win) {
+            initMediaQueryListeners(win);
+        } else {
+            updateBreakpoints.call(win);
+            win.addEventListener('resize', debounce(updateBreakpoints, 200));
+        }
+    }
+
+    function initMediaQueryListeners(win) {
+        breakpoints
+            .forEach(function (bp, index, bps) {
+                // We create mutually exclusive (min-width) and (max-width) media queries
+                // to facilitate the breakpoint/tweakpoint logic.
+                bp.mql = index < bps.length - 1 ?
+                    win.matchMedia('(min-width:'+ bp.width +'px) and (max-width:'+ (bps[index+1].width - 1) +'px)') :
+                    win.matchMedia('(min-width:'+ bp.width +'px)');
+                bp.listener = onMatchingBreakpoint.bind(bp);
+                bp.mql.addListener(bp.listener);
+                bp.listener(bp.mql);
+            });
+    }
+
+    function onMatchingBreakpoint(mql) {
+        if (mql.matches) {
+            updateBreakpoint(this);
+        }
+    }
+
+    function updateBreakpoint(breakpoint) {
+        if (breakpoint.isTweakpoint) {
+            currentTweakpoint = breakpoint.name;
+            currentBreakpoint = findBreakpoint(currentTweakpoint);
+        } else {
+            currentBreakpoint =
```

Comment:
this is a bit unclear i think – took me a second to work out what was happening. how do you feel about putting them on one line?